harvester_e2e_tests/integration/test_images.py

```diff
@@ -1,47 +1,385 @@
+
+        resp = api_client.images.create_by_file(unique_name, fake_image_file)
+
+        assert 200 == resp.status_code, (
+            f"Failed to upload fake image with error:{resp.status_code}, {resp.content}"
+        )
+
+        get_image(api_client, unique_name)
+        delete_image(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    def test_create_invalid_file(self, api_client, unique_name,
+                                 fake_invalid_image_file, wait_timeout):
+        """
+        Test create upload image from invalid file type
+
+        Steps:
+        1. Prepare an invalid file that is not in a multiple of 512 bytes
+        2. Try to upload invalid image file which to images page
+        2. Check should get an error
+        """
+
+        resp = api_client.images.create_by_file(unique_name, fake_invalid_image_file)
+
+        assert 500 == resp.status_code, (
+            f"File size correct, it's a multiple of 512 bytes:{resp.status_code}, {resp.content}"
+        )
+        delete_image(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="edit_image_in_use", depends=["create_image_url"])
+    def test_edit_image_in_use(self, api_client, unique_name, qcow2_name,
+                               iso_name, wait_timeout):
+        """
+        Test can edit image which already in use
+
+        Steps:
+        1. Check the image created from URL exists
+        2. Create a volume from existing image
+        3. Update the image labels and description
+        4. Check can change the image content
+        """
+
+        get_image(api_client, qcow2_name)
+
+        code, data = api_client.images.get()
+        assert 200 == code, (code, data)
+
+        for item in data['items']:
+            if item['spec']['displayName'] == qcow2_name:
+                image_id = item['metadata']['uid']
+
+        # Create volume from image_id
+        spec = api_client.volumes.Spec(1)
+        code, data = api_client.volumes.create(unique_name, spec, image_id=image_id)
+
+        assert 201 == code, (code, data)
+
+        # Check volume ready
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.volumes.get(unique_name)
+            if data["status"]["phase"] == "Bound":
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to delete volume {unique_name} bound in {wait_timeout} timed out\n"
+                f"Still got {code} with {data}"
+            )
+
+        get_volume(api_client, unique_name)
+
+        # Update image
+        updates = {
+            "labels": {
+                "usage-label": "yes"
+            },
+            "annotations": {
+                "field.cattle.io/description": 'edit image in use'
+            },
+
+        }
+
+        update_by_input(api_client, qcow2_name, updates)
```

Comment:
It looks `update_by_input` will only be called once, same as `get_export`.

To notice each test case is isolated and independent, when creating a helper function, it should not have any side effects (like assertion inside), otherwise we will need to debug the helper function rather than the test case; and when we update the helper function to fix the error, it will affect other test cases which calling the function.

Reply:
Thanks for the review, move the function content of `update_by_input` and `get_export` to the test case blocks